terraform/backend.tf

```diff
@@ -0,0 +1,6 @@
+terraform {
```

Comment:
Could we add more documentation around the TF module?
* A reference in the Python README to prepare the infra needed by going to this module
* A README in the TF module to 
* * Create the backend bucket
* * demonstrate the command line usage (e.g. TF commands, variables substitution, etc)